src/measure_theory/interval_integral.lean

```diff
@@ -209,66 +276,778 @@ lemma integral_add_adjacent_intervals (hfm : measurable f) (hab : interval_integ
+  refine (this.congr_left _).trans_is_O _,
+  { intro x, simp [sub_smul] },
+  { exact is_O_fst_prod.norm_left.add is_O_snd_prod.norm_left }
+end
+
+/-- Fundamental theorem of calculus-1: if `f : ℝ → E` is integrable on `a..b` and `f` is continuous
+at `a` and `b`, then `(u, v) ↦ ∫ x in u..v, f x` has derivative `(u, v) ↦ v • cb - u • ca`
+at `(a, b)` in the sense of strict differentiability. -/
+lemma integral_has_strict_fderiv_at (hfm : measurable f)
+  (hfi : interval_integrable f volume a b) (ha : continuous_at f a) (hb : continuous_at f b) :
+  has_strict_fderiv_at (λ p : ℝ × ℝ, ∫ x in p.1..p.2, f x)
+    ((snd ℝ ℝ ℝ).smul_right (f b) - (fst ℝ ℝ ℝ).smul_right (f a)) (a, b) :=
+integral_has_strict_fderiv_at_of_tendsto_ae hfm hfi
+  (ha.mono_left inf_le_left) (hb.mono_left inf_le_left)
+
+/-- Fundamental theorem of calculus-1: if `f : ℝ → E` is integrable on `a..b` and `f x` has a finite
+limit `c` almost surely at `b`, then `u ↦ ∫ x in a..u, f x` has derivative `c` at `b` in the sense
+of strict differentiability. -/
+lemma integral_has_strict_deriv_at_of_tendsto_ae_right (hfm : measurable f)
+  (hfi : interval_integrable f volume a b) (hb : tendsto f (𝓝 b ⊓ volume.ae) (𝓝 c)) :
+  has_strict_deriv_at (λ u, ∫ x in a..u, f x) c b :=
+integral_sub_integral_sub_linear_is_o_of_tendsto_ae_right hfm hfi hb continuous_at_snd
+  continuous_at_fst
+
+/-- Fundamental theorem of calculus-1: if `f : ℝ → E` is integrable on `a..b` and `f` is continuous
+at `b`, then `u ↦ ∫ x in a..u, f x` has derivative `f b` at `b` in the sense of strict
+differentiability. -/
+lemma integral_has_strict_deriv_at_right (hfm : measurable f)
+  (hfi : interval_integrable f volume a b) (hb : continuous_at f b) :
+  has_strict_deriv_at (λ u, ∫ x in a..u, f x) (f b) b :=
+integral_has_strict_deriv_at_of_tendsto_ae_right hfm hfi (hb.mono_left inf_le_left)
+
+/-- Fundamental theorem of calculus-1: if `f : ℝ → E` is integrable on `a..b` and `f x` has a finite
+limit `c` almost surely at `a`, then `u ↦ ∫ x in u..b, f x` has derivative `-c` at `a` in the sense
+of strict differentiability. -/
+lemma integral_has_strict_deriv_at_of_tendsto_ae_left (hfm : measurable f)
+  (hfi : interval_integrable f volume a b) (ha : tendsto f (𝓝 a ⊓ volume.ae) (𝓝 c)) :
+  has_strict_deriv_at (λ u, ∫ x in u..b, f x) (-c) a :=
+by simpa only [← integral_symm]
+  using (integral_has_strict_deriv_at_of_tendsto_ae_right hfm hfi.symm ha).neg
+
+/-- Fundamental theorem of calculus-1: if `f : ℝ → E` is integrable on `a..b` and `f` is continuous
+at `a`, then `u ↦ ∫ x in u..b, f x` has derivative `-f a` at `a` in the sense of strict
+differentiability. -/
+lemma integral_has_strict_deriv_at_left (hfm : measurable f)
+  (hfi : interval_integrable f volume a b) (ha : continuous_at f a) :
+  has_strict_deriv_at (λ u, ∫ x in u..b, f x) (-f a) a :=
+by simpa only [← integral_symm] using (integral_has_strict_deriv_at_right hfm hfi.symm ha).neg
+
+/-!
+#### Unbounded differentiability
```

Comment:
I don't understand the term `unbounded` here.

Reply:
Me too. I guess I meant "without `_within`". Removed.